### DISCUSSION TITLE
What is this syntax in the tutorial?

### DISCUSSION THREAD

**Discussion by fnick851** [remix-run/remix#1673] [Q&A] [answered]
In this tutorial: https://remix.run/docs/en/v1/tutorials/blog
```js
function isValidPostAttributes(
  attributes: any
): attributes is PostMarkdownAttributes {
  return attributes?.title;
}
```

What is the `is` keyword? - `attributes is PostMarkdownAttributes`? Wow, I never saw something like that before. Is this not a typo?

**Comment by AlexandrLi** [comment, marked as answer]
This is a user-defined type guard. You can read more about it in docs https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

**Reply by fnick851** [reply]
Thank you.
